### ISSUE TITLE
Gigabyte GA-Z170M-D3H with QQLS CPU modded bios can't boot after adding ReBarDxe.ffs

### ISSUE THREAD

**Issue by Igo131** [xCuri0/ReBarUEFI#115]
**System**
 - Motherboard: Gigabyte ga-z170m-d3h
 - BIOS Version: F22f (modded for QQLS cpu)
 - GPU: QQLS
 - GPU: rx6700
 - [x] CSM is turned off. Checked in bios.
 - [x] 4G decoding is enabled. Checked in bios.
 - [ ] UEFIPatch and DSDTPatch are not applied.
 - [x] I have read [Common issues (and fixes)](https://github.com/xCuri0/ReBarUEFI/wiki/Common-issues-(and-fixes))

Description

Hi. I have a modded bios file from QQLS supplier to work with this cpu. Added ReBarDxe.ffs in UEFITool_0.28.0. After flashing with q-flash or ch341 board can't load (no post, bootloop). No image from internal/external gpu (somehow i managed to boot once but unabled to enable Rebar as well). Normally all loads even with csm and 4g on/off. Tried MMTool but can't find PciBus string there.

![2024-01-15 195515](https://github.com/xCuri0/ReBarUEFI/assets/156717270/b041ed3d-3f23-49ab-be99-37f682a7f975)
![2024-01-16 025551](https://github.com/xCuri0/ReBarUEFI/assets/156717270/ff3ddd22-ee5b-4e0e-9bd5-9bbcef4e0eb0)

Only this in Large Memory:
![2024-01-16 044256](https://github.com/xCuri0/ReBarUEFI/assets/156717270/07bb508c-7a4e-4b7a-8a20-fb388c068ca2)

[Z170M-D3H F22f-ReBAR.zip](https://github.com/xCuri0/ReBarUEFI/files/13943254/Z170M-D3H.F22f-ReBAR.zip)
[Z170M-D3H F22f.zip](https://github.com/xCuri0/ReBarUEFI/files/13943253/Z170M-D3H.F22f.zip)

**Comment by takem94** [comment]
I don’t quite understand, after flashing the BIOS with the ReBarState module, were you able to enter the OS or BIOS? I looked at your BIOS via coffeetime 0.99, there are a lot of unnecessary patches and microcodes and there is no patch for PCIE.

![image](https://github.com/xCuri0/ReBarUEFI/assets/23483056/c160db08-c230-4ffc-9cdf-55e3270830a3)

It may be worth rebuilding the BIOS, the ME version is probably too new, although it is turned off, it takes up a lot of space.

If you still manage to get into the system with the ReBar module and its value at 0, 8, 9 or even 10. It may be useful to look at my topic https://github.com/xCuri0/ReBarUEFI/issues/114 .

**Comment by Igo131** [comment]
[takem94](https://github.com/takem94)
Thank you for the reply.

The issue was in the modded bios from qqls seller. Modded clean bios:

![2024-01-16 234016](https://github.com/xCuri0/ReBarUEFI/assets/156717270/b1af4cba-d3b4-48b7-8ec2-3360432251e3)
![2024-01-16 234050](https://github.com/xCuri0/ReBarUEFI/assets/156717270/882b7f7e-f771-4646-ba0f-155b12759c6b)

Resizeble Bar works now.

**Comment by takem94** [comment]
@Igo131 you can do this patch: 
`# PciBus | Don't downgrade 64-bit BARs to 32-bit (Coffee Lake)
3C1DE39F-D207-408A-AACC-731CFB7F1DD7 10 P:A80274334038AF95:A802EB334038AF95`
But they all seem to be unnecessary. 

In theory, it should be possible to get into the BIOS via the iGPU, at least for me, I connect by hdmi cable.
With High Definition Audio Bus, as I understand it is the same problem for everyone https://github.com/xCuri0/ReBarUEFI/issues/4 or most, I use a separate sound card in any case, so it’s not critical for me.

**Comment by Igo131** [comment]
[takem94](https://github.com/takem94) Seems like the patch is not needed or am I patching it wrong:

![2024-01-17 034030](https://github.com/xCuri0/ReBarUEFI/assets/156717270/e097aa62-367b-436d-8176-16809ac3c754)

I saw you benchmarks with 1gb rebar. Maybe 1gb will be enouth for games performance boost if there is no fix for High Definition Audio Bus problem. What you think?

**Comment by xCuri0** [comment]
@takem94 these are usual issue with LGA1151 and resizable bar, there's already an issue about this. It's weird considering older Sandy/Ivy Bridge and Haswell/Broadwell have zero issues like this.

**Comment by Igo131** [comment]
> @xCuri0 as i see there is no fix for now?
Anyway if someone will do some patches i can test it on my system (have ch341).

**Comment by takem94** [comment]
@Igo131 I think it's very individual and depends on many factors. Firstly, I have a card from Nvidia and with a small VRAM size, which is almost always full of game data, so there is very little cache left for other memory operations and apparently 1 GB is the limit. Also important is the game itself and how it uses Rebar. You need to do tests in main games for yourself and decide worth it or not.

**Comment by xCuri0** [comment]
@takem94 Nvidia driver only uses 1GB BAR in most games anyways unless you use NVIDIA Inspector to manually increase it. Sometimes it results in fps loss though

AMD always uses the full BAR size but even then in my experience the difference between 2GB and 8GB was negligible for me even tho 256MB vs 2GB was an almost 15% increase

**Comment by takem94** [comment]
@xCuri0 yeah, thank you. I did not know about the limitation at the driver level and used Rebar only through the inspector, because with a value of 1 gigabyte, the driver panel indicates that Rebar is disabled. 
Also, on Intel cards without Rebar support, many games are on borderline of playability. I think for them your work cannot be overestimated. 
It's still hard to believe that this works on my old platform and even older ones. It’s a pity, of course, that new games are increasingly greedy for processor power, even CS2 on a full server in DM with an analogue i7 8700 I can get an average of 120 frames and 1 percent on 70. I played CS:GO and COD all my life and always had about 200-300 FPS with more bad CPUs. Now I hope to at least buy a Ryzen 7500f to get the same result.
